src/pymovements/measure/measures.py

```diff
@@ -50,11 +52,111 @@ def null_ratio(column: str, column_dtype: pl.DataType) -> pl.Expr:
         value = 1 - pl.col(column).count() / pl.col(column).len()
     elif column_dtype == pl.List:
         non_null_lengths = pl.col(column).list.drop_nulls().drop_nans().list.len()
-        value = 1 - (non_null_lengths == pl.col(column).list.len()).sum() / pl.col(column).len()
+        value = (
+            1 - (non_null_lengths == pl.col(column).list.len()).sum() /
+            pl.col(column).len()
+        )
     else:
         raise TypeError(
             'column_dtype must be of type {Float64, Int64, Utf8, List}'
             f' but is of type {column_dtype}',
         )
 
     return value.alias('null_ratio')
+
+
+@register_sample_measure
+def data_loss(
+        timestamp_col: str = 'time',
+        *,
+        start_time: float | None = None,
+        end_time: float | None = None,
+        unit: Literal['count', 'time', 'ratio'] = 'count',
+) -> pl.Expr:
+    """Measure gaze data loss within a group using only Polars expressions.
+
+    Computes the number of missing samples ("count"), missing time ("time"),
+    or the ratio of missing to expected samples ("ratio"). The expected
+    sampling is derived from the group's median inter-sample interval (ISI)
+    computed on the ``timestamp_col`` after sorting.
+
+    If ``start_time``/``end_time`` are not provided, the group's first/last
+    timestamps are used (i.e., ``min``/``max`` of ``timestamp_col``).
+
+    Parameters
+    ----------
+    timestamp_col: str
+        Name of the timestamp column. Should be numeric and (ideally) monotonic
+        within the group.
+    start_time: float | None
+        Recording start time for the group. If ``None``, uses the group's first timestamp.
+    end_time: float | None
+        Recording end time for the group. If ``None``, uses the group's last timestamp.
+    unit: Literal['count', 'time', 'ratio']
+        - "count": number of missing samples (absolute)
+        - "time": missing time in timestamp units (approx. ``missing_count * median_ISI``)
+        - "ratio": ``missing_count / expected`` (in [0, 1])
+
+    Returns
+    -------
+    pl.Expr
+        A scalar (per-group) expression.
+
+    Raises
+    ------
+    ValueError
+        If ``unit`` is not one of 'count', 'time', or 'ratio'.
+    TypeError
+        If ``timestamp_col`` is not a string.
+    """
+    if not isinstance(timestamp_col, str):
+        raise TypeError(
+            f'timestamp_col must be a string, but got: {type(timestamp_col).__name__}',
+        )
+
+    timestamps = pl.col(timestamp_col)
```

Comment:
let's use ducktyping here with a `try`/`except` block:

```suggestion
    try:
        timestamps = pl.col(time_column)
    except TypeError as e:
       raise TypeError(
           f"invalid type for 'time_column'. Expected 'str' , got '{type(time_column).__name__}'",
        ) from e
```

This has two benefits:

- we may adapt with less changes in case of a broader input typing of the called 3rd party function
- the error trace for the failing case is not obfuscated from the user
- there is no if condition inference (marginal in this case)